dashpay/src/main/kotlin/org/dashevo/dashpay/Profiles.kt

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2020-present, Dash Core Group
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package org.dashevo.dashpay
+
+import org.bouncycastle.crypto.params.KeyParameter
+import org.dashevo.platform.Platform
+
+class Profiles(
+    val platform: Platform, private val blockchainIdentity: BlockchainIdentity,
+    private val keyParameter: KeyParameter
+) {
```

Comment:
This is fine for now, but when we add a `get()` method there is no need for a `blockchainIdentity` object or a keyParameter for queries.  Those queries will require an identity id (string).  

In the other classes (Names, Identities, Documents) they are rather primitive and don't know if a BlockchainIdentity object is involved.  Instead `create` type methods take an identity id (string), or other more primitive data types for the Identity and keys and key parameters.  I am not sure if these DashPay `documents` type classes like `Profiles` and `ContactRequests` should be dependent on `BlockchainIdentity` or ignorant of `BlockchainIdentity` - where instead `BlockchainIdentity` uses these classes to perform dashpay related operations. 

In the long term, keyParameter shouldn't be saved in this class.  Mainly for security reasons.  I think that is why in DashJ, individual methods that sign data take a key parameter (the encryption key) as an argument.  Perhaps after we do the crypto course we can discuss and address these security concerns.  What happens before any DPP calls are made is that the encrypted keys and decrypted and those decrypted keys are sent to the dpp library to do the actual signing.  Not sure if that is a good idea either, it was a way to support encrypted keys without changing `android-dpp`.